# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLPictureElement.cpp
# Commit: acdd013202e1
# Full Hash: acdd013202e1e688986e7e6c747b7c7fee38474e
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1694741
# File Overlap Count: 1
# Description:
#   Bug 1694741 - Part 6: Map width/height attributes to the style of img elements. r=emilio
#   
#   This patch will use the width/height attributes from <source> to override
#   width/height/aspect-ratio CSS property values of <img> elements.
#   
# ==============================================================================

diff -r 2b4d9516b882 -r acdd013202e1 dom/html/HTMLPictureElement.cpp
--- a/dom/html/HTMLPictureElement.cpp	Mon Aug 22 20:18:39 2022 +0000
+++ b/dom/html/HTMLPictureElement.cpp	Mon Aug 22 20:18:39 2022 +0000
@@ -60,7 +60,7 @@
     img->PictureSourceAdded();
   } else if (auto* source = HTMLSourceElement::FromNode(aKid)) {
     // Find all img siblings after this <source> to notify them of its insertion
-    nsCOMPtr<nsIContent> nextSibling = aKid->GetNextSibling();
+    nsCOMPtr<nsIContent> nextSibling = source->GetNextSibling();
     if (nextSibling && nextSibling->GetParentNode() == this) {
       do {
         if (auto* img = HTMLImageElement::FromNode(nextSibling)) {